Tests/UnitTests/Core/Propagator/CovarianceEngineTests.cpp

```diff
@@ -114,8 +114,7 @@ BOOST_AUTO_TEST_CASE(covariance_engine_test) {
   auto curvResult = detail::curvilinearState(
       covariance, jacobian, transportJacobian, derivatives, boundToFreeJacobian,
       parameters, particleHypothesis, false, 1337.);
-  BOOST_CHECK(std::get<0>(curvResult).covariance().has_value());
-  BOOST_CHECK_EQUAL(*(std::get<0>(curvResult).covariance()), covarianceBefore);
+  BOOST_CHECK(!std::get<0>(curvResult).covariance().has_value());
```

Comment:
Can we move the declaration of `auto covarianceBefore = covariance;` down to line 138, where we reset `covarianceBefore = covariance;`?